frontend/src/components/Bundle/Bundle.js

```diff
@@ -62,114 +62,80 @@ class Bundle extends React.Component<
+        };
+
+        fetchBundleMetadata(this.props.uuid)
+            .then(callback)
+            .catch(errorHandler);
 
         // Fetch bundle contents
-        $.ajax({
-            type: 'GET',
-            url: '/rest/bundles/' + this.props.uuid + '/contents/info/',
-            data: {
-                depth: 1,
-            },
-            dataType: 'json',
-            cache: false,
-            context: this, // automatically bind `this` in all callbacks
-        })
-            .then(async function(response) {
-                const info = response.data;
-                if (!info) return;
-                if (info.type === 'file' || info.type === 'link') {
-                    return this.fetchFileSummary(this.props.uuid, '/').then(function(blob) {
-                        this.setState({ fileContents: blob, stdout: null, stderr: null });
-                    });
-                } else if (info.type === 'directory') {
-                    // Get stdout/stderr (important to set things to null).
-                    let fetchRequests = [];
-                    let stateUpdate = {
-                        fileContents: null,
-                    };
-                    ['stdout', 'stderr'].forEach(
-                        function(name) {
-                            if (info.contents.some((entry) => entry.name === name)) {
-                                fetchRequests.push(
-                                    this.fetchFileSummary(this.props.uuid, '/' + name).then(
-                                        function(blob) {
-                                            stateUpdate[name] = blob;
-                                        },
-                                    ),
-                                );
-                            } else {
-                                stateUpdate[name] = null;
-                            }
-                        }.bind(this),
-                    );
-                    await Promise.all(fetchRequests);
-                    this.setState(stateUpdate);
-                }
-            })
-            .fail(
-                function(xhr, status, err) {
-                    // 404 Not Found errors are normal if contents aren't available yet, so ignore them
-                    if (xhr.status !== 404) {
-                        this.setState({
-                            bundleInfo: null,
-                            fileContents: null,
-                            stdout: null,
-                            stderr: null,
-                            errorMessages: this.state.errorMessages.concat([xhr.responseText]),
-                        });
-                    } else {
-                        // If contents aren't available yet, then also clear stdout and stderr.
-                        this.setState({ fileContents: null, stdout: null, stderr: null });
-                    }
-                }.bind(this),
-            );
+        callback = async (response) => {
```

Comment:
same comment as above

Reply:
actually, not this one, it's used twice

Reply:
The two functions are to handle two different API calls here I think.